lisa/platform_.py

```diff
@@ -108,6 +128,9 @@ def prepare_environment(self, environment: Environment) -> Environment:
         """
         self.initialize()
 
+        platform_message = PlatformMessage(name=self.type_name())
```

Comment:
Add a flag to make sure it's called once. The `prepare_environment` will be called multiple times, if multiple environment is necessary.

Reply:
This message will be sent with a different name on each initialization. This would be useful to take different actions for different environments.

Reply:
it means the platform is initialized, not an environment. If we need track the environment lifecycle, we can create messages for environments.

Reply:
On a second thought I think I mixed up environment and platform. Is a platform initialized for every environment?

Reply:
no, the platform is initialized once for each run. The platform creates one or more environments.

Reply:
In that case shouldn't we include `environment` as part of `PlatformMessage`? It would a holistic message and could be used to track lifecycle of different environments. And if we want to just check if a platform is created, we can use `INITIALIZED`  status which will only be called once.

Reply:
Well, to support current requirement, we can create only initialized message for platform. That's enough. For environment lifecycle, we can leave it, and create it later.

Reply:
Sounds good! I will remove the other message and just leave the `INITIALIZED ` message.

Reply:
I don't think that will work. Initialise message is called inside `prepare_environment` and that would be called multiple times for each environment. This also means that `initialize` is being redundantly called multiple times for each environment. Let me think of a better way on this.

Reply:
How about we remove `initialize` call from `prepare_environment` and call it explicitly after a platform is declared? For example, we call `platform.initialise()` after platform declaration in `lisa_runner.py`. This will ensure that initialise is called only once for each platform. https://github.com/microsoft/lisa/blob/60d8fa432379c2603f90542a21886fd37a4b9892/lisa/runners/lisa_runner.py#L56

Reply:
yes, good idea.

Reply:
But even the initialize is in prepare, we can have a flag to make sure it's called only once.

Reply:
I am suggesting to remove the initialize in prepare environment method. And call it explicitly after initializing platform.